### PR TITLE
Handle rerequested checks by scheduling computation again

### PR DESCRIPTION
**Pull request by lukebjerring**
## Description
Cleans up and collapses two code branches that (should) do the same thing.
Now when a check_suite (or run) is rerequested, we add the check(s) to the TaskQueue for computing the outcome.

### CODE REVIEW

File: api/checks/webhook.go [hunk 1]
```diff
@@ -212,18 +212,19 @@ func handlePullRequestEvent(ctx context.Context, payload []byte) (bool, error) {
...
 		return false, fmt.Errorf("Failed to load test runs: %s", err.Error())
 	}
 	createdSome := false
+	api := NewSuitesAPI(ctx)
 	for _, rbp := range runsByProduct {
 		if len(rbp.TestRuns) > 0 {
-			created, err := completeCheckRun(ctx, sha, rbp.Product)
-			createdSome = createdSome || created
+			err := api.ScheduleResultsProcessing(sha, rbp.Product)
+			createdSome = createdSome || err == nil
```
Comment (Hexcles) [inline api/checks/webhook.go]: Hmm, I don't think you need `createdSome` any more. In fact, this method probably only need to return `error` now that you defer the processing to TaskQueue.

File: api/checks/webhook.go [hunk 2]
```diff
@@ -266,19 +268,10 @@ func createCheckRun(ctx context.Context, suite shared.CheckSuite, opts github.Cr
...
-	}
-	req.Header.Set("Accept", "application/vnd.github.antiope-preview+json")
-	checkRun := new(github.CheckRun)
-	resp, err := client.Do(ctx, req, checkRun)
 
+	checkRun, resp, err := client.Checks.CreateCheckRun(ctx, suite.Owner, suite.Repo, opts)
 	if err != nil {
 		log.Warningf("Failed to create check_run: %s", resp.Status)
-		body, _ := ioutil.ReadAll(resp.Body)
-		log.Warningf(string(body))
```
Comment (Hexcles) [inline api/checks/webhook.go]: Is this logging not useful?

File: api/taskcluster/webhook_test.go [hunk 3]
```diff
@@ -130,6 +130,9 @@ func TestCreateAllRuns_success(t *testing.T) {
 
 	mockC := gomock.NewController(t)
 	suitesAPI := checks.NewMockSuitesAPI(mockC)
+	suitesAPI.EXPECT().GetSuitesForSHA(sha).Return([]shared.CheckSuite{
+		shared.CheckSuite{SHA: sha},
+	}, nil)
 	suitesAPI.EXPECT().PendingCheckRun(sha, "chrome")
```
Comment (Hexcles) [inline api/taskcluster/webhook_test.go]: Hmm, how did this pass? `PendingCheckRun` now takes a suite. Did you forget to regenerate the mock?

Comment (lukebjerring) [reply api/taskcluster/webhook_test.go]: Actually, it takes an array of suites to update... so "this is fine". I'll update the test :)

File: api/checks/webhook.go (hunk 2 above)
Comment (lukebjerring) [reply api/checks/webhook.go]: No, it was a debugging effort that didn't get cleaned up.

File: api/checks/webhook.go (hunk 1 above)
Comment (lukebjerring) [reply api/checks/webhook.go]: It affects the content of the HTTP response (as to whether anything was scheduled or not).

**Pull request merged**